cli/popper/parser.py

```diff
@@ -79,25 +79,35 @@ def get_stages(self):
                         'next', set()))
             current_stage = next_stage
 
-    def complete_graph(self):
+    def complete_graph(self, action_name=None):
```

Comment:
can we update the docstring so it explains the purpose of this new parameter? Is this new parameter restricting how deep the completion goes? I feel like the code is a bit difficult to parse. 

Conceptually, it might be more readable to let the `complete_graph` work on the complete graph and then implement any sort of filtering or ad-hoc graph traversal separately. So in this case, we could leave the `complete_graph` function as it was before, and instead add an `action` parameter to the `get_stages` method that indicates that we want to traverse the graph from the root to the given action (as opposed to going from the root to the end of the graph), ignoring any nodes that are not a direct dependency to this given action

Reply:
Okay